### PR TITLE
[GPT-15] Pagination for search

### PR DESCRIPTION
**Pull request by gajinkee**
Added the pagination. I think it looks pretty good.
Also fixed a bug where clearing input with the x button doesn't reset the search.

ss1
ss2
ss3

### CODE REVIEW

**Comment by wei2912** [comment]
1. Remove pagination bar when empty, have visual indicator to disable buttons if there's no left/right page
2. Vertically align "Showing … results" with the left/right buttons.
3. On mobile, the "Showing … results" and left/right buttons should still be on the same row.
4. Remove scrolling where possible (but for mobile it's fine)
5. Increase font size and modal window size for desktop, fix modal size

**Comment by gajinkee** [comment]
Made all the changes mentioned. Works better for mobile now too. Will demo at the next meeting.

ss1
ss2

File: assets/css/site/modal.css [hunk 1]
```diff
@@ -23,7 +23,9 @@
 }
 
 .modal-search {
-    width: 500px;
+    width: 60%;
+    height: 54%;
```
Comment (wei2912) [inline assets/css/site/modal.css:27]: Can we increase the height further to 80%?

**Review by wei2912 — CHANGES_REQUESTED** [review]
Have left some minor remarks. Besides the below comments,
1. Could the pagination buttons be greyed out and disabled if there is no prev/next page of results?
2. Ideally the scrollbar for the results should not appear, and the height of the results modal should be fixed (so that the positioning of the searchbar and buttons don't change for each page).

I will review the JS code closely when the above changes have been made.

**Comment by wei2912** [comment]
Functionally, I think this is ready for merge. There are a few other issues to be addressed to improve the design:

1. I think there needs to be some more changes made to ensure the search feature is responsive on mobile:

image

2. Longer post titles are not handled well:

image

3. 7 is a strange number for no. of results, 5 may work better.

**Comment by wei2912** [comment]
(3) has been addressed in https://github.com/AdvisorySG/dawn-advisory-theme/pull/228/commits/8d886a3012074672e715b64b8d8fd2448379678d.

**Pull request merged**